### ISSUE TITLE
Error on slate watch, cloud-hosted files / Dropbox

### ISSUE THREAD

**Issue by kevmon** [Shopify/slate#315]
Encountering an error similar to one that motivated me to reformat.  I've read around a little and think it might be related to permissions and gulp but I don't really understand it yet.  After reformatting, I installed NVM and Node through that, using my admin account in OSX.  Not really sure how to troubleshoot.

Kevins-MacBook-Pro:FlySupplyMe kevin$ slate watch -n
module.js:529
    throw err;
    ^

Error: Cannot find module '../lib/completion'
    at Function.Module._resolveFilename (module.js:527:15)
    at Function.Module._load (module.js:476:23)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/kevin/Sync/Work/Sites/FlySupplyMe/node_modules/.bin/gulp:13:18)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
Kevins-MacBook-Pro:FlySupplyMe kevin$

### Replication steps

- Fresh OS High Sierra reformat and install
- Node Version Manager v0.33.5
- Node v8.6.0
- NPM v5.4.2
- Slate v0.12.1

Also, I don't know if this is relevant, but I keep all of my sites in my sync.com folder, which is constantly syncing across multiple computers.

Thanks in advance

**Comment by kevmon** [comment]
Update:  I seem to have narrowed down the problem to the node_modules folder within the project itself.  I created a new slate theme and moved my src files into it, and it works fine.  I'm not sure how, but that folder must have become corrupted at some point.  Perhaps it's due to my cloud syncing service?

**Comment by mattsince87** [comment]
@kevmon i have a _similar issue_ #320 and tried doing this. It just resulted with me receiving more errors (Failed to preview due to missing theme files; which were not missing). I think *Slate* is still very new and is absolutely **not** suitable for production at this point. Previously i used ThemeKit with Grunt to watch, compile and preprocess all my js,scss and liquid. The only thing i wished for which *is* available is a better local dev environment with a local server, the rest of slate is dreadful.

**Comment by hanihsdd** [comment]
I have also encountered this issue on multiple occasions when I run `slate watch`. Similar to @kevmon, I work across multiple computers and have my Slate project in Dropbox. 

I think it's an issue with Gulp and syncing to the cloud. [This post on Stack Overflow](https://stackoverflow.com/questions/36081200/cannot-find-module-lib-completion-inspite-of-installing-completion) provides a simple solution to the problem which is re-installing gulp.

Uninstall:
`npm uninstall gulp --save`

Re-Install:
`npm install gulp --save`

That quickly solves the issue. Though it would be great to find out what the problem is and stop it from happening.

**Comment by NathanPJF** [comment]
@kevmon and @hanihsdd Thanks for providing the additional information that your files are being hosted on with a cloud service/Dropbox.  

I spent some time reading on the Internet and asking around the office, but I couldn't find a silver bullet for solving issues with `node_modules` and cloud-based file hosting.  Though I did find a bunch of posts of other people facing this issue with other CLI tools 😑 

I wonder if an alternative for uninstalling/re-installing `gulp`, would be to have selective sync for Dropbox ignore your node_modules.  So you'd have to run `npm install` once on each computer, but there'd be no syncing activity on that folder that would potentially corrupt those files.

I'm going to rename this issue to reflect hosted files so it's a bit more descriptive to people who may encounter this in the future.

---

@mattsince87 It sounds like your previous development setup was better suited for how you want to work.  I've responded to your other issue https://github.com/Shopify/slate/issues/320 in this regard
